src/core/mount.c

```diff
@@ -1718,23 +1736,38 @@ static int mount_setup_unit(
         if (set_flags)
                 MOUNT(u)->proc_flags = flags;
 
+        *mu = MOUNT(u);
+
         return 0;
 }
 
 static int mount_load_proc_self_mountinfo(Manager *m, bool set_flags) {
         _cleanup_(mnt_free_tablep) struct libmnt_table *table = NULL;
         _cleanup_(mnt_free_iterp) struct libmnt_iter *iter = NULL;
         int r;
+        static bool mountinfo_hashmap_initialized = false;
 
         assert(m);
 
+        if (!mountinfo_hashmap_initialized) {
+                r = hashmap_ensure_allocated(&mountinfo_hashmap, &string_hash_ops);
```

Comment:
hashmap_ensure_allocated() checks if the hashmap is allocated, so this conditional and tracking variable are not needed.

Reply:
ok